===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ PQ_Single:n$2 == { -1 } }) free(*(PQ_Single:pro)) at 5 (line 989, column 9) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/snort-2.9.13/src/snort.c
+++ b/table1/benchmarks/snort-2.9.13/src/snort.c
@@ -987,6 +987,7 @@ static void PQ_Single (const char* pcap)
     pro->filter = NULL;
 
     if (sflist_add_tail(pcap_object_list, (NODE_DATA)pro) == -1)
+        free(pro);
         FatalError("Could not add pcap object to list: %s\n", pcap);
 }
 
